components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -592,7 +617,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ApplicationError(ErrorCodes.NOT_AUTHENTICATED, "User is not authenticated. Please login.");
         }
 
-        const user = await this.userService.findUserById(this.userID, this.userID);
+        const user = await this.userService.findUserById(SYSTEM_USER, this.userID);
```

Comment:
why do you treat this as a system call? I think this is only called based on a user action.

Reply:
I see `checkUser` as something that we, the system, want to perform to validate the user is actually allowed to interact with it, and enforce other invariants. This should be independent of any permissions the user has or not has.